reactive/kotlinx-coroutines-reactive/src/Channel.kt

```diff
@@ -75,6 +74,7 @@ private class SubscriptionChannel<T>(
     @Suppress("CANNOT_OVERRIDE_INVISIBLE_MEMBER")
     override fun onClosedIdempotent(closed: LockFreeLinkedListNode) {
         subscription?.cancel()
+        subscription = null // optimization -- no need to cancel it again
```

Comment:
Please use CAS instead to have a more consistent guarantee.
E.g. reactor does exactly the same, see `Operators.terminate`